python/interpret_community/mimic/mimic_explainer.py

```diff
@@ -133,14 +134,19 @@ class MimicExplainer(BlackBoxExplainer):
     :param reset_index: Uses the pandas DataFrame index column as part of the features when training
         the surrogate model.
     :type reset_index: str
+    :param auto_select_explainable_model: Set this to 'True' if you want to use the MimicExplainer with an
```

Comment:
I wonder if this should be a separate explainer or function - mimic explainer takes a specific surrogate model and not a list.  This also seems like something that complicates mimic explainer logic.  Maybe we can discuss more.

Thinking of other libraries, usually there is a distinction between hyperparameter tuning and training (eg in both v1 studio and designer there is a Train Model and Tune Hyperparameters or Cross validate module, in spark ML the hyperparameter tuner is a separate estimator, in scikit-learn similarly grid search cv is a separate function).  I feel like for users who want to do this we should have a separate function/class instead of complicating the current mimic explainer.